pkg/virt-handler/vm.go

```diff
@@ -377,6 +389,36 @@ func domainMigrated(domain *api.Domain) bool {
 	return false
 }
 
+func (d *VirtualMachineController) getPodInterfacefromFileCache(vmi *v1.VirtualMachineInstance, ifaceName string) (*network.PodCacheInterface, error) {
+	ifacepath := fmt.Sprintf(virtutil.VMIInterfacepath, vmi.ObjectMeta.UID, ifaceName)
+
+	// Once the Interface files are set on the handler, they don't change
+	// If already present in the map, don't read again
+	d.podInterfaceCacheLock.Lock()
+	result, exists := d.podInterfaceCache[ifacepath]
+	d.podInterfaceCacheLock.Unlock()
+
+	if exists {
+		return result, nil
+	}
+
+	content, err := ioutil.ReadFile(ifacepath)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to read from cache file: %s", err.Error())
+		return nil, err
+	}
+	err = json.Unmarshal(content, &result)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to unmarshal interface content: %s", err.Error())
+		return nil, err
+	}
+	d.podInterfaceCacheLock.Lock()
+	d.podInterfaceCache[ifacepath] = result
```

Comment:
i think you're already fixing this.  cache entry needs to be deleted when the cache file is deleted.

Reply:
Corrected now